contractkit: use @celo/abis package

shazarre opened the pull request:
### Description

This PR aims to introduce `@celo/abis` package usage to contractkit.

### Other changes

Changes the way `@celo/abis` package is generated.

### Tested

By using the package in kit

### Related issues

- Fixes #10719

### Backwards compatibility

* import paths changed in @celo/abis from `@celo/abis/types/web3 => @celo/abis/web3`, 

* import paths changed in @celo/abis from `@celo/abis/types/wagmi => @celo/abis`, 

* @celo/contractkit no longer has `lib/generated` folder. (these are now all in `@celo/abis/web3`

### Documentation

updated readme

shazarre committed:
revert ESM/CJS compilation, use previous directory structure

shazarre committed:
merge web3/* folders into one as well

shazarre committed:
change working dir to packages/protocol/abis/src-generated

aaronmgdr commented on packages/protocol/scripts/prepare-contracts-and-abis-publishing.ts:
```diff
@@ -67,42 +68,23 @@ try {
-  })
+  const file = fs.readFileSync(sourcePackageJson).toString()
+
+  if (process.env.RELEASE_VERSION) {
```
doing some validity check on this would probably be wise. I think SemVer package is already used in the script?

shazarre committed:
include interfaces

shazarre committed:
prepare contracts package.json

shazarre committed:
fix package.json not exisiting for @celo/abis package

aaronmgdr committed:
fix npm tagging, and a spelling issue

shazarre committed:
leave nested dirs in web3/ generated folder

shazarre committed:
Merge branch 'shazarre/contractkit_use_celo_abis_package' of github.com:celo-org/celo-monorepo into shazarre/contractkit_use_celo_abis_package

aaronmgdr committed:
fix #10564 warning

aaronmgdr committed:
fix #10564 warning

shazarre committed:
add @celo/abis usage and change imports

shazarre committed:
Merge branch 'shazarre/contractkit_use_celo_abis_package' of github.com:celo-org/celo-monorepo into shazarre/contractkit_use_celo_abis_package

shazarre committed:
change @celo/abis to be a regular dependency instead of a dev one

shazarre committed:
remove extraneous fields from generated jsons

shazarre committed:
missing yarn.lock + bump up @celo/abis

aaronmgdr committed:
no longer need build:gen as the contract interfaces come from celo/abis... so also no need to clean /generated.

shazarre committed:
esm/cjs compilation

shazarre committed:
missed changed imports

shazarre committed:
contract cache: use @celo/abis package

shazarre committed:
proxy: use @celo/abis package

shazarre committed:
mini contract cache: use @celo/abis package

shazarre committed:
contract cache: use @celo/abis package

shazarre committed:
replace all other usages of ./generated

aaronmgdr committed:
fix other places import must change and add changeset

shazarre committed:
use @celo/abis/types/web3 instead of generated/

shazarre committed:
bump up @celo/abis version

aaronmgdr committed:
abis are imported to these packages

aaronmgdr committed:
must set moduleResolution to node16,

should be targeting at least es6

aaronmgdr committed:
keep declarations separate

aaronmgdr committed:
yolo

aaronmgdr committed:
two experiments,
dont build all packages before running the script as it will build what it needs
always prepare

aaronmgdr committed:
we dont need to change target as it already set.

aaronmgdr committed:
may it be so

aaronmgdr committed:
must run packages protocol needs, then prepare abis, then contractkit ++

aaronmgdr committed:
dont need to cache these

aaronmgdr committed:
order imports, and fix esModuleInterpo issue

shazarre committed:
fix linting errors

aaronmgdr committed:
fix failure made noticable by test

aaronmgdr committed:
lint cli

aaronmgdr committed:
Merge branch 'master' into shazarre/contractkit_use_celo_abis_package

aaronmgdr committed:
fix ledger connection

aaronmgdr committed:
clean up ts configs

aaronmgdr committed:
target must be es5 for tests but es2020 for export

aaronmgdr commented:
https://github.com/celo-org/celo-monorepo/pull/10781 should mean the error in the SortedOracles ts tests no longer matters

aaronmgdr committed:
es5 is a must

shazarre committed:
remove types/ prefix

shazarre committed:
change imports

shazarre committed:
comment out temporarily

shazarre committed:
Merge branch 'master' into shazarre/contractkit_use_celo_abis_package

shazarre committed:
is this the changed file? + uncomment

shazarre committed:
Merge branch 'shazarre/contractkit_use_celo_abis_package' of github.com:celo-org/celo-monorepo into shazarre/contractkit_use_celo_abis_package

shazarre committed:
add packages/protocol/abis/src-generated/ to .prettierignore

shazarre committed:
lint fix

shazarre committed:
remove ethers specific files generation

aaronmgdr committed:
esm should be esm

aaronmgdr committed:
better changeset

aaronmgdr committed:
use rc of abi

aaronmgdr committed:
shhhh - we are already not expecting this to be working on current version so no worries

aaronmgdr committed:
exclude source maps

aaronmgdr committed:
generate ts files at top level

aaronmgdr committed:
fix the readme and undo some temp stuff

therealharpaljadeja committed:
feat(@celo/abis): add Node env usage (#10797)

* feat(@celo/abis): add Node env usage

* docs(README): improves examples

- adds `viem` example in JS/TS section (previously missing)
- simplifies CommonJS example

---------

Co-authored-by: 0xarthurxyz <46296830+0xarthurxyz@users.noreply.github.com>
Co-authored-by: Aaron DeRuvo <aaron@clabs.co>

aaronmgdr committed:
use latest versions

aaronmgdr committed:
fix mento tokens missing from the abis , add index.js

aaronmgdr committed:
dont publish contract tests

aaronmgdr committed:
need actual file location

aaronmgdr committed:
give esm consumers a index entry

shazarre committed:
update READMEs

shazarre committed:
Merge branch 'shazarre/contractkit_use_celo_abis_package' of github.com:celo-org/celo-monorepo into shazarre/contractkit_use_celo_abis_package

aaronmgdr committed:
rc 4

shazarre committed:
update yarn.lock

aaronmgdr committed:
ignore

shazarre committed:
copy 0.8 contracts to the package

shazarre committed:
Merge branch 'shazarre/contractkit_use_celo_abis_package' of github.com:celo-org/celo-monorepo into shazarre/contractkit_use_celo_abis_package

shazarre committed:
missed consts file

shazarre committed:
@celo/abis: rc4 -> rc5

aaronmgdr committed:
gold master
